state-chain/runtime/src/chainflip/bitcoin_block_processor.rs

```diff
@@ -423,14 +352,16 @@ mod tests {
 		type BlockData = MockBlockData;
 		type Event = MockBtcEvent;
 		type Rules = ApplyRulesHook;
-		type Execute = IncreasingHook<Self::Event, ()>;
-		type CleanOld = CleanOldBlockDataHook;
+		type Execute = IncreasingHook<(Self::ChainBlockNumber, Self::Event), ()>;
 		type DedupEvents = DedupEventsHook;
+
+		fn get_safety_margin() -> Self::ChainBlockNumber {
```

Comment:
I think this should be a hook, then we can give it a random value in prop tests.